### PR TITLE
[release-4.6] Bug 1898096: Ensure correct git URL for topology edit icon

### PR DESCRIPTION
**Pull request by rottencandy**
Manual cherry-pick of #7158 and #7216

### CODE REVIEW

**Comment by abhinandan13jan** [comment]
overall looks good

**Comment by rottencandy** [comment]
> overall looks good
> Need to fix the associated bug though

It's waiting for #7158 to be verified.

File: frontend/packages/dev-console/src/components/topology/topology-utils.ts [hunk 1]
```diff
@@ -35,9 +36,14 @@ export const getServiceBindingStatus = ({ FLAGS }: RootState): boolean =>
 export const getCheURL = (consoleLinks: K8sResourceKind[]) =>
   _.get(_.find(consoleLinks, ['metadata.name', 'che']), 'spec.href', '');
 
-export const getEditURL = (gitURL: string, gitBranch: string, cheURL: string) => {
+export const getEditURL = (vcsURI?: string, gitBranch?: string, cheURL?: string) => {
+  if (!vcsURI) {
+    return null;
+  }
```
Comment (andrewballantyne) [inline frontend/packages/dev-console/src/components/topology/topology-utils.ts:42]: @rottencandy Why did this become needed in the z-stream but not in the 4.7 version?

**Review by andrewballantyne — CHANGES_REQUESTED** [review]

File: frontend/packages/dev-console/src/components/topology/topology-utils.ts (hunk 1 above)
Comment (rottencandy) [reply frontend/packages/dev-console/src/components/topology/topology-utils.ts:42]: This change is in 4.7 as well, with #7216

Comment (andrewballantyne) [reply frontend/packages/dev-console/src/components/topology/topology-utils.ts:42]: Ah, gotcha! Misunderstood the description when I took a quick look at this.

**Review by andrewballantyne — APPROVED** [review]
/lgtm

**Pull request merged**